src/ucs/datastruct/bitmap.h

```diff
@@ -63,8 +162,72 @@ typedef uint64_t ucs_bitmap_word_t;
 #define _UCS_BITMAP_DECLARE_TYPE(_length) \
     typedef struct { \
         ucs_bitmap_word_t bits[_UCS_BITMAP_BITS_TO_WORDS(_length)]; \
-    } ucs_bitmap_t(_length);
-
+    } ucs_bitmap_t(_length); \
+\
+    static inline bool ucs_bitmap_##_length##_is_zero(ucs_bitmap_t(_length) \
+                                                          bitmap) \
+    { \
+        return UCS_BITMAP_IS_ZERO(bitmap); \
+    } \
+\
+    /** \
+     * Perform bitwise NOT of a bitmap \
+     * \
+     * @param _bitmap Negate this bitmap \
+     * \
+     * @return A new bitmap, which is the negation of the given one \
+     */ \
+    static inline ucs_bitmap_t(_length) \
+        ucs_bitmap_##_length##_not(ucs_bitmap_t(_length) bitmap) \
```

Comment:
why do we need these inline functions, if already have macors?

Reply:
As described in the PR explanation, we distinguish between "in-place" ops, and those that return a value.
The "in-place" ops are macros and they are (IMHO) the preferred way of performing bit ops on a bitmap.
The reason is that they allow the user to *not* specify the exact bitmap type, thus if at some future day TL bitmap changes from 256bit to 512bit, such code:
```
UCS_BITMAP_AND_INPLACE(tl_bitmap, bitmap2);
```
does not require change, which is cool.

But sometimes we *can't* make "in-place" ops - when we only have an r-value.
Consider the following code (in master):
```
 ucs_assert(!(ucp_ep_get_tl_bitmap(ep) & ~local_tl_bitmap));
```
One option is to assign to a variable, then do it in-place:
```
tmp = ucp_ep_get_tl_bitmap(ep);
tmp2 = local_tl_bitmap;
UCS_BITMAP_NOT_INPLACE(tmp2);
UCS_BITMAP_AND_INPLACE(tmp, tmp2);
ucs_assert(UCS_BITMAP_IS_ZERO(tmp);
```
IMHO it is bad option - the code looks ugly, many temporary vars and hard to read and follow it.
A better (but not ideal) solution is using these inline functions (that's what I wrote them for):
```
ucs_assert(ucs_bitmap_128_is_zero(ucs_bitmap_128_and(
           ucp_ep_get_tl_bitmap(ep), ucs_bitmap_128_not(*local_tl_bitmap))));
```
The advantage is that the code looks much more readable than in the first option, and very similar to the original `&`, `~` syntax. The disadvantage is having to mention explicitly the bitmap type, thus upon change it will have to be updated.

Reply:
Notice that from compilation perspective, these forms are identical. Copying will anyway take place exactly one time, and can't be avoided.

Reply:
let's try to add length in the bitmap struct to enable using functions which return bitmap by value, but can also see the bitmap length. those could be generic inline functions to replace all macros.

Reply:
Done, as we discussed